kernel_gateway/services/notebooks/handlers.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import tornado.web
+import json
+from queue import Empty
+from .request_utils import *
+
+class NotebookAPIHandler(tornado.web.RequestHandler):
+    get_source = None
+    put_source = None
+    post_source = None
+    delete_source = None
+    kernel_client = None
+    execution_timeout = 5
+    def initialize(self, sources, kernel_client):
```

Comment:
IIRC `initialize` is invoked per instance of this class in tornado, and *every web request spawns its own instance*. That's why the handler map is to classes, not instances.

Definitely don't want to be doing this parsing over and over again. Best to keep these out of the class at the module level (singleton) or as class vars.